### PR TITLE
zxcc: add livecheck and update license

### PR DESCRIPTION
**Pull request by mprince**
```sh
$ brew livecheck zxcc -d
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/zxcc.rb

Formula:          zxcc
Livecheckable?:   Yes

URL:              https://www.seasip.info/Unix/Zxcc/
Strategy:         PageMatch
Regex:            /href=.*?zxcc[._-]v?(\d+(?:\.\d+)+)\.t/i

Matched Versions:
0.5.7, 0.4.0
zxcc : 0.5.7 ==> 0.5.7
```

### CODE REVIEW

**Review by samford — COMMENTED** [review]
The `livecheck` block looks good to me 👍

I'm not a license expert, so I'll leave that to someone with more experience. From a cursory glance, there appear to be various source files (in `cpmio/`, `cpmredir`, etc.) that have license information at the top that mention "version 2 of the License, or (at your option) any later version". I'm not sure how those subdirectories apply to how we build `zxcc` but I only mention it insomuch as I think the `license` change needs supporting evidence and review.

**Comment by mprince** [comment]
@samford I think you are right. I was looking at the `COPYING` file and did not see anything mentioning a later version. I would change it to include `-or-later` while awaiting review.

**Review by SMillerDev — APPROVED** [review]

**Comment by samford** [comment]
> I was looking at the `COPYING` file and did not see anything mentioning a later version.

@mprince The `COPYING` file only contains the GPLv2 license text, which unfortunately doesn't tell you whether it's `-only` or `-or-later`. It's necessary to look elsewhere (upstream website, documentation, source files, etc.) for a brief explanation of how the license applies. This [previous PR](https://github.com/Homebrew/homebrew-core/pull/63795), where another maintainer explained this concept to me, may help you to better understand this.

In this case, I checked the first-party website and it only said:

* "The libraries are LGPLed and should be fairly easy to use separately from ZXCC."
* "This version (0.5.7) is GPLed, and uses the GNU Autotools to build."

This doesn't give us enough information to even determine that it's GPLv2.

Next, I downloaded the latest tarball, unarchived it, and checked the `COPYING` file, which at least makes it clear this is GPLv2. Next, I `grep`'d the files for things like "later", "GPL", etc., as this gives me an idea of if any of the source files contain license comments at the top. Some of the included libraries that `zxcc` uses have explicit licenses at the top of the source files that make it clear they're `GPL-2.0-or-later`. For example, `bin/z80.c` has the following comment:

```
/* Emulation of the Z80 CPU with hooks into the other parts of xz80.
 * Copyright (C) 1994 Ian Collier.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
```

Text like `either version 2 of the License, or (at your option) any later version.` is what we're usually looking for. [To be clear, the similar text in the `COPYING` file isn't an indicator and only functions to explain that the license allows this type of application. The text of how the license is applied is found elsewhere, as mentioned above.]

That said, the source files for `zxcc` itself (e.g., `bin/zxcc.c`) don't contain any license information like this. I didn't look into this in depth but I didn't find any clear indicator of how `zxcc` itself is licensed with regard to `-only` or `-or-later`. I'm also not sure how `GPL-2.0-or-later` libraries interact with the GPLv2 license of the software that uses it (e.g., if `zxcc` must also be `GPL-2.0-or-later` to use `GPL-2.0-or-later` libraries).

Someone with more experience with licenses and a better understanding of how they are applied can explain this more accurately but hopefully this is helpful. `zxcc` is clearly GPLv2 but it's still unclear to me whether it's `GPL-2.0-or-later`. That said, this has been approved/merged at this point, so that's that. Thanks!

**Comment by SMillerDev** [comment]
I didn't spot that there was still an open issue about the license here. If there's any uncertainty the license should be removed @mprince. Could you make a new PR for that?

**Pull request closed without merging**
